src/ceph-disk/ceph_disk/main.py

```diff
@@ -3153,6 +3153,61 @@ def move_mount(
     )
 
 
+def systemd_enable(
+    path,
+    osd_id,
+):
+    # ensure there is no duplicate ceph-osd@.service
+    for style in ([], ['--runtime']):
+        command_check_call(
+            [
+                'systemctl',
+                'disable',
+                'ceph-osd@{osd_id}'.format(osd_id=osd_id),
+            ] + style,
```

Comment:
thanks. i sort of understand now: the enabled `--runtime` units and enabled non-runtime units are living in different naming spaces, literally `/run` and `/etc`. in other words, we are not able to disable both of them in a single shot without `--runtime`.